### PR TITLE
Prevent panic on /remote.php/dav/files/

### PR DESCRIPTION
**Pull request by refs**
Currently requests to `/remote.php/dav/files/` result in panics since we cannot longer strip the user + destination from the url.

We're reproducing this issue on the ocis storage. I wanted to tackle this at the storage level but sadly the panic happens before reaching the stat call.

I wonder if there is a better way to achieve the same results but instead letting the storage handle the way it interprets paths. Open to suggestions 💃

### CODE REVIEW

**Comment by labkode** [comment]
@refs can you provide info of the panic?

**Comment by phil-davis** [comment]
https://cloud.drone.io/cs3org/reva/3145/13/6
```
runsh: Total unexpected passed scenarios throughout the test run:
apiWebdavOperations/propfind.feature:5
```

This does "a good thing" - that now-passing scenario should be deleted from the expected-failures files.

**Comment by phil-davis** [comment]
And look at https://cloud.drone.io/cs3org/reva/3145/6/7 tests/acceptance/features/apiOcisSpecific/apiShareWebdavOperations-propfind.feature:6

```
  Scenario: PROPFIND to "/remote.php/dav/files"                                            # /drone/src/tests/acceptance/features/apiOcisSpecific/apiShareWebdavOperations-propfind.feature:6
    Given user "Alice" has been created with default attributes and without skeleton files # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
    When user "Alice" requests "/remote.php/dav/files" with "PROPFIND" using basic auth    # AuthContext::userRequestsURLWithUsingBasicAuth()
    Then the HTTP status code should be "500"                                              # FeatureContext::thenTheHTTPStatusCodeShouldBe()
      HTTP status code 405 is not the expected value 500
      Failed asserting that 405 matches expected '500'.
```

That bug-demo scenario is now producing a good result. It can be deleted.

**Comment by refs** [comment]
hi @labkode sorry for the radio silence, I was off sick the past week with some string children virus thing ...

The panic I was referring to happens here:

https://github.com/cs3org/reva/blob/b2c4af4e897d3890e3f7e476547875e384babf8d/internal/grpc/services/gateway/storageprovider.go#L2017-L2020

and it is triggered when the URL looks just like in the description of this pull request:

`/remote.php/dav/files/`

I'm open to better alternatives and not just patch this :)

**Comment by labkode** [comment]
@refs to trigger this condition one has to be on a "Shared folder" that must contain these structure.
If the path is outside the Shared folder (remote.php/dav/files) is not a shared folder this method should never be called.
Is it possible that your config is missing or has a different value for the Shares folder?

Anyways, the panic could have some nice logging to inform of the problem.

**Review by butonic — CHANGES_REQUESTED** [review]

File: internal/http/services/owncloud/ocdav/propfind.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ func (s *svc) handlePropfind(w http.ResponseWriter, r *http.Request, ns string)
 		case rpc.Code_CODE_PERMISSION_DENIED:
 			log.Debug().Str("path", fn).Interface("status", res.Status).Msg("permission denied")
 			w.WriteHeader(http.StatusMultiStatus)
+		// TODO this is not a correct mapping. A new code SHOULD be added to the cs3apis.
+		case rpc.Code_CODE_UNIMPLEMENTED:
```
[the remark this answers is not in the document]
Comment (refs) [reply internal/http/services/owncloud/ocdav/propfind.go:99]: agree, will contain the changes to ocdav only

**Comment by refs** [comment]
@butonic changes are now contained to `ocdav.go` and its path handling magic 🌸 will put some effort into returning the correct XML and refactoring a little bit.

I thought of a middleware for not allowed methods to prevent from executing unnecessary logic but it might be an early optimization, but the fact that there is no way to catch this early on our router has some room for improvement 🚀 

@labkode as per the behavior, it is captured here: https://github.com/owncloud/core/blob/18475dac812064b21dabcc50f25ef3ffe55691a5/tests/acceptance/features/apiWebdavOperations/propfind.feature

honestly that LoC on Reva was running regardless of being on a shared folder or not, the stack trace shed some light on the path taken, I recommend having a look at it. I will dig into my configuration to discard it but we're not having a very complex use case, but we need to be sure.

**Review by butonic — APPROVED** [review]

File: tests/acceptance/features/apiOcisSpecific/apiShareWebdavOperations-propfind.feature [hunk 2]
```diff
@@ -6,4 +6,4 @@ Feature: PROPFIND
   Scenario: PROPFIND to "/remote.php/dav/files"
```
Comment (butonic) [inline tests/acceptance/features/apiOcisSpecific/apiShareWebdavOperations-propfind.feature:6]: If you check the line above you should find the ` after fixing all issues delete this Scenario and use the one from oC10 core`, meaning that AFAICT this complete feature file can be removed, as it is covered by the core tests.

- [ ] remove `tests/acceptance/features/apiOcisSpecific/apiShareWebdavOperations-propfind.feature`

**Review by butonic — CHANGES_REQUESTED** [review]

**Review by phil-davis — COMMENTED** [review]
Looks good from a test point of view - the related scenario in the core API tests now passes, and the local bug-demo scenario has been deleted.

**Review by butonic — APPROVED** [review]

**Comment by ishank011** [comment]
@refs can we try to return false instead of raising a panic here? 
https://github.com/cs3org/reva/blob/b2c4af4e897d3890e3f7e476547875e384babf8d/internal/grpc/services/gateway/storageprovider.go#L2017-L2020

The next step will be to find the storage provider and since it won't find any mapping for it, it will return the appropriate error. That's because this is raised a lot of other times when the user doesn't provide a correct path from any other clients.

**Comment by refs** [comment]
> @refs can we try to return false instead of raising a panic here?
> 
> https://github.com/cs3org/reva/blob/b2c4af4e897d3890e3f7e476547875e384babf8d/internal/grpc/services/gateway/storageprovider.go#L2017-L2020
> 
> The next step will be to find the storage provider and since it won't find any mapping for it, it will return the appropriate error. That's because this is raised a lot of other times when the user doesn't provide a correct path from any other clients.

I'd say that's out of the scope of this one :D

**Review by ishank011 — APPROVED** [review]
Oh okay. No worries!

**Pull request merged**